### DISCUSSION TITLE
code size reduced for nrf52840dk

### DISCUSSION THREAD

**Discussion by nehalrohit** [zephyrproject-rtos/zephyr#30576] [Q&A]
Hello,
I merged  the code  updatehub+openthread+5 I2c sensor+SDcard for nrf52840dk_pac10056 (all extra logs are diasble).
My problem is the code(bin file ) size is more than **0x67000**  having any reason why zephyr-based code(simple CLI code -412 KB) takes more space?
how to reduce the code size?

**Comment by ndrs-pst** [comment]
Try to set CONFIG_DEBUG_OPTIMIZATIONS=y or CONFIG_SIZE_OPTIMIZATIONS=y if you haven't set before.
Also using rom_report could help you find out the big part of your code.
-> https://docs.zephyrproject.org/latest/develop/optimizations/tools.html
